core/src/main/java/org/apache/accumulo/core/spi/balancer/TabletBalancer.java

```diff
@@ -67,6 +67,15 @@ interface AssignmentParameters {
      * Assigns {@code tabletId} to {@code tabletServerId}.
      */
     void addAssignment(TabletId tabletId, TabletServerId tabletServerId);
+
+    /**
+     * Balancers can use this mapping in conjunction with {@link Property#TABLE_ASSIGNMENT_GROUP} to
+     * assign tablets to tablet servers within the corresponding resource group
+     *
+     * @return map of resource group name to set of TServerInstance objects
+     * @since 4.0.0
+     */
+    Map<String,Set<TabletServerId>> currentResourceGroups();
```

Comment:
This would be a follow on, we may want to introduce a type for the group names instead of using string.  This would be something like TableId.

```suggestion
    Map<String,Set<TabletServerId>> currentResourceGroups();
```

Reply:
The suggestion looks the same to me, but I think I understand what you are saying.

Reply:
I think I meant to put some type name in the suggestion.  Must have gotten distracted by something while writing the comment.